Content.Client/Chat/ChatManager.cs

```diff
@@ -33,37 +49,62 @@ public void SetChatBox(ChatBox chatBox)
             if (_currentChatBox != null)
             {
                 _currentChatBox.TextSubmitted -= _onChatBoxTextSubmitted;
+                _currentChatBox.FilterPressed -= _onFilterButtonToggled;
             }
 
             _currentChatBox = chatBox;
             if (_currentChatBox != null)
             {
                 _currentChatBox.TextSubmitted += _onChatBoxTextSubmitted;
+                _currentChatBox.FilterPressed += _onFilterButtonToggled;
             }
         }
 
         private void _onChatMessage(MsgChatMessage message)
         {
             Logger.Debug($"{message.Channel}: {message.Message}");
 
-            var color = Color.DarkGray;
-            var messageText = message.Message;
-            if (!string.IsNullOrEmpty(message.MessageWrap))
+            if (filteredHistory.ContainsValue(message))
             {
-                messageText = string.Format(message.MessageWrap, messageText);
+                return;
             }
 
-            switch (message.Channel)
+            // Set time message sent
+            message.TimeStamp = DateTime.Now;
```

Comment:
If you intend to just keep these messages sorted by the order in which they come in you should just keep them in a regular `List<T>`. Using a `SortedDictionary` and `DateTime.Now` is an overly roundabout and inaccurate way to do it.